tests-arquillian/src/test/java/org/jboss/weld/tests/instance/enhanced/InstanceHandleTest.java

```diff
@@ -0,0 +1,129 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.weld.tests.instance.enhanced;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import jakarta.enterprise.context.Dependent;
+import jakarta.enterprise.inject.Instance;
+import jakarta.enterprise.inject.spi.Bean;
+import jakarta.enterprise.inject.spi.BeanManager;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.BeanArchive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.weld.test.util.ActionSequence;
+import org.jboss.weld.test.util.Utils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Iterator;
+import java.util.List;
+
+@RunWith(Arquillian.class)
+public class InstanceHandleTest {
+
+    @Deployment
+    public static Archive<?> createTestArchive() {
+        return ShrinkWrap.create(BeanArchive.class, Utils.getDeploymentNameAsHash(InstanceHandleTest.class))
+                .addClasses(Alpha.class, Bravo.class, Client.class, FirstProcessor.class, Processor.class,
+                        SecondProcessor.class, Juicy.class)
+                .addPackage(WeldInstanceTest.class.getPackage())
+                .addClass(ActionSequence.class);
+    }
+
+    @Test
+    public void testIsResolvable(Client client) {
+        ActionSequence.reset();
+        assertNotNull(client);
+        assertTrue(client.getAlphaInstance().isResolvable());
+        assertFalse(client.getBigDecimalInstance().isResolvable());
+    }
+
+    @Test
+    public void testGetHandler(Client client, BeanManager beanManager) {
+        ActionSequence.reset();
+        assertNotNull(client);
+
+        Bean<?> alphaBean = beanManager.resolve(beanManager.getBeans(Alpha.class));
+        Instance<Alpha> instance = client.getAlphaInstance();
+
+        Instance.Handle<Alpha> alpha1 = instance.getHandle();
+        assertEquals(alphaBean, alpha1.getBean());
+        assertEquals(Dependent.class, alpha1.getBean().getScope());
+
+        String alpha2Id;
+
+        // Test try-with-resource
+        try (Instance.Handle<Alpha> alpha2 = instance.getHandle()) {
+            alpha2Id = alpha2.get().getId();
+            assertFalse(alpha1.get().getId().equals(alpha2Id));
+        }
+
+        List<String> sequence = ActionSequence.getSequenceData();
+        assertEquals(1, sequence.size());
+        assertEquals(alpha2Id, sequence.get(0));
+
+        alpha1.destroy();
+        // Subsequent invocations are no-op
+        alpha1.destroy();
+
+        sequence = ActionSequence.getSequenceData();
+        assertEquals(2, sequence.size());
+        assertEquals(alpha1.get().getId(), sequence.get(1));
```

Comment:
This is actually wrong - we cannot assert the ID at this point as the instance was already destroyed. We need to change the behavior to instead throw an exception when you perform `Handle.get()` in a `Handle` that destroyed its reference. Plus we need to test that (in both test variants).